python/cudf/cudf/tests/test_groupby.py

```diff
@@ -1749,13 +1755,19 @@ def test_groupby_shift_row_mixed_numerics(
     expected = pdf.groupby(["0"]).shift(periods=n_shift, fill_value=fill_value)
     got = gdf.groupby(["0"]).shift(periods=n_shift, fill_value=fill_value)
 
-    assert_groupby_results_equal(expected, got)
+    # Pandas returns shifted column in original row order. We set its index
+    # to be the key columns, so that `assert_groupby_results_equal` can sort
+    # rows by key columns to make sure cudf and pandas results matches.
+    expected.index = gdf["0"].to_pandas()
```

Comment:
should we change this index in the test, or should we be returning the same order as pandas in the public api?

Reply:
Currently `groupby.agg` already diverts from pandas in row orders, in that cudf does not alter the key order returned from libcudf to match pandas order. Here we follow the same convention.